posthog/api/plugin.py

```diff
@@ -0,0 +1,132 @@
+    class Meta:
+        model = Plugin
+        fields = ["id", "name", "description", "url", "config_schema", "tag", "from_json"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
+        if len(Plugin.objects.filter(name=validated_data["name"])) > 0:
+            raise APIException('Plugin with name "{}" already installed!'.format(validated_data["name"]))
+        validated_data["archive"] = download_plugin_github_zip(validated_data["url"], validated_data["tag"])
+        if "from_json" in validated_data:  # prevent hackery
+            del validated_data["from_json"]
+        plugin = Plugin.objects.create(from_web=True, **validated_data)
+        reload_plugins_on_workers()
+        return plugin
+
+    def update(self, plugin: Plugin, validated_data: Dict, *args: Any, **kwargs: Any) -> Plugin:  # type: ignore
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
+        if plugin.from_json:
+            raise APIException('Can not update plugin "{}", which is configured from posthog.json!'.format(plugin.name))
+        plugin.name = validated_data.get("name", plugin.name)
+        plugin.description = validated_data.get("description", plugin.description)
+        plugin.url = validated_data.get("url", plugin.url)
+        plugin.config_schema = validated_data.get("config_schema", plugin.config_schema)
+        plugin.tag = validated_data.get("tag", plugin.tag)
+        plugin.archive = download_plugin_github_zip(plugin.url, plugin.tag)
+        plugin.save()
+        reload_plugins_on_workers()
+        return plugin
+
+
+class PluginViewSet(viewsets.ModelViewSet):
+    queryset = Plugin.objects.all()
+    serializer_class = PluginSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        if not settings.PLUGINS_INSTALL_FROM_WEB and not settings.PLUGINS_CONFIGURE_FROM_WEB:
+            return queryset.none()
+        return queryset
+
+    @action(methods=["GET"], detail=False)
+    def repository(self, request: request.Request):
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            return Response([])
+        url = "https://raw.githubusercontent.com/PostHog/plugins/main/repository.json"
+        plugins = requests.get(url)
+        return Response(json.loads(plugins.text))
+
+    def destroy(self, request: request.Request, pk=None) -> Response:  # type: ignore
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
+        plugin = Plugin.objects.get(pk=pk)
+        if plugin.from_json:
+            raise APIException('Can not delete plugin "{}", which is configured from posthog.json!'.format(plugin.name))
+        plugin.delete()
+        reload_plugins_on_workers()
```

Comment:
I would suggest changing this method to just
```python
response = super().destroy(request, *args, **kwargs)
reload_plugins_on_workers()
return response
````

Reply:
Fixed! Kept the raises above though.